corehq/apps/locations/expressions.py

```diff
@@ -0,0 +1,20 @@
+from corehq.apps.userreports.specs import TypeProperty
+from dimagi.ext.jsonobject import JsonObject
+
+
+class LocationTypeSepc(JsonObject):
+    type = TypeProperty('location_type_name')
+
+    def __call__(self, item, context=None):
+        try:
+            doc_id = item['_id']
+            from corehq.apps.locations.models import SQLLocation
+            sql_location = SQLLocation.objects.filter(location_id=doc_id)
```

Comment:
we should cache the look up of ID to type so that this doesn't hit the DB multiple times while processing a single form. The easiest way to do that would be to make it its own function and use `quickcache`